### ISSUE TITLE
Cannot touch retrieved session tokens at 'signIn' event on OAuth

### ISSUE THREAD

**Issue by si0005hp** [aws-amplify/amplify-js#5925]
**Describe the bug**
On social provider signin (OAuth signin), we can't touch retrieved session tokens when capturing `signIn` event (via `Hub`) since they are set to the client side storage **after** the event.
While on normal signIn case, they are set **before** the event and so we can touch on them.

**To Reproduce**
**Code Snippet**

```js
import React from "react"
import { Authenticator, SignIn, ForgotPassword } from "aws-amplify-react"
import { Auth, Hub } from "aws-amplify"
import { getMemoryStorage } from "./MyStorage"

const memoryStorage = getMemoryStorage()

Hub.listen("auth", (data) => {
  switch (data.payload.event) {
    case "signIn":
      /*
       * Here, in case of
       *  - Social provider signin: -> session tokens are NOT set
       *  - Normal signin: -> session tokens are set
       *  on both 'memoryStorage' and 'data.payload'.
       */
      doSomethingUsingSessionToken(memoryStorage)
      break
    default:
  }
})

/* ↓ Not the essence ↓  */
export const App: React.FC = () => (
  <>
    <Authenticator hideDefault={true}>
      <SignIn />
      <ForgotPassword />
    </Authenticator>
  </>
)
const oauthConfig = {
  domain: "mydomain.amazoncognito.com",
  scope: ["email", "openid", "aws.cognito.signin.user.admin"],
  redirectSignIn: "http://localhost:3000/callback",
  redirectSignOut: "http://localhost:3000/logout",
  responseType: "code"
}
Auth.configure({
  region: "ap-northeast-1",
  userPoolId: "XXXXXXXXXX",
  userPoolWebClientId: "XXXXXXXXXX",
  authenticationFlowType: "USER_PASSWORD_AUTH",
  oauth: oauthConfig,
  storage: memoryStorage
})
```

(There's no special setting and it happens on very basic condition of each login flow.)

**Expected behavior**
Hope session tokens are set on the storage at the signIn event in both (in any of) cases.

**Context (of expectaion)**
At the `doSomethingUsingSessionToken` in the code above, we have basically 2 things to be done.

1. Save session tokens (on `memoryStorage`) to server

+ To avoid forcing users re-login whenever they reflesh the page, we also keep session tokens on  server side. (The approach is almost the same as [this](https://docs.amplify.aws/lib/auth/manageusers/q/platform/js#managing-security-tokens). )
  ※The reason why we use memory (avoiding localStorage which is default of amplify) on client side is the security concern.

2. Verify the user information using retrieved idToken

+ We permit the social login not to everyone but to only our specific customers, and so we need a kind of validation thing using retrieved idToken.

**Additional context**
Here are the brief description of our app stuck.

- SPA with its own backend (API) server
- Using cognito user pool for user account data
- Using amplify-js for auth logic handling and UI on client side

**Comment by si0005hp** [comment]
@elorzafe Hi, thanks for paying attention to this.
Seems you've fixed the problem in this [PR #5929](https://github.com/aws-amplify/amplify-js/pull/5929/files#diff-1cf5b10c87e919f81d5f1ea503f15074R1907) ?
(In that case I guess I can close this issue and my [PR](https://github.com/aws-amplify/amplify-js/pull/5927). )

**Comment by si0005hp** [comment]
@elorzafe Hi, sorry for many ping and bothering, but today I've upgraded to the version `3.0.14` (which includes your fix #5929) and confirmed the problem has gone.
Thank you so much for your work! it's helpful for us :pray:.
I'll close this issue and #5927 (should be no longer necessary).
